### PR TITLE
Update developer guide with important information.

### PR DESCRIPTION
**Pull request by mikewiebe**
These are modifications that are needed in advance of the developer training next week.

### CODE REVIEW

File: docs/README-develop-node-utils-APIs.md [hunk 1]
```diff
@@ -3,32 +3,48 @@
...
  * [Step 1. YAML Definitions: router eigrp](#comp_yaml)
  * [Step 2. Create the node_utils API: router eigrp](#comp_api)
  * [Step 3. Create the Minitest: router eigrp](#comp_minitest)
  * [Step 4. rubocop / lint: router eigrp](#comp_lint)
+ * [Step 5. Build the gem](#comp_gem)
 
 ## <a name="overview">Overview</a>
 
-This document is a HowTo guide for writing new cisco node_utils APIs. The node_utils APIs act as an interface between the NX-OS CLI and an agent's resource/provider. If written properly the new API will work as a common framework for multiple providers (Puppet, Chef, etc).
+This document is a HowTo guide for writing new cisco-network-node-utils APIs. The APIs act as an interface between the NX-OS CLI and an agent's resource/provider. If written properly the new API will work as a common framework for multiple providers (Puppet, Chef, etc).
```
Comment (glennmatthews) [inline docs/README-develop-node-utils-APIs.md]: Not sure about this change and the one below - while the project name is cisco-network-node-utils, the Gem name is cisco_node_utils.

File: docs/README-develop-node-utils-APIs.md [hunk 2]
```diff
@@ -3,32 +3,48 @@
...
-First install the code base. Clone the cisco_node_utils repo into a workspace:
+This development guide uses tools that are packaged as gems that need to be installed on your workstation.
 
 ```bash
-git clone https://github.com/cisco/cisco-network-node-utils.git
+gem install cisco_nxapi
+gem install rubocop
+gem install minitest --version 4.3.2
+```
+
```
Comment (glennmatthews) [inline docs/README-develop-node-utils-APIs.md:38]: Do we also need to walk people through installing git-flow? (https://github.com/petervanderdoes/gitflow-avh/wiki#installing-git-flow)

Also, if the training will have people working from an engineering server where they don't have admin/root privileges, you might want to include a note about using `gem install --user` and adding `~/.gem/ruby/x.x.x./bin` to their `PATH` - or you can move this step after the 'git clone' step and suggest using bundler instead to install the gems into the local repo copy (`bundle install --path ./vendor/bundle`)

File: docs/README-develop-node-utils-APIs.md [hunk 3]
```diff
@@ -3,32 +3,48 @@
...
+## <a name="clone">Start here: Fork and Clone the Repo</a>
+
+First [fork](https://help.github.com/articles/fork-a-repo) the [cisco-network-node-utils](https://github.com/cisco/cisco-network-node-utils) git repository 
+
+Next install the code base. Clone the cisco-network-node-utils repo from your fork into a workspace:
+
+```bash
+git clone https://github.com/YOUR-USERNAME/cisco-network-node-utils.git
+cd cisco-network-node-utils/
 ```
```
Comment (glennmatthews) [inline docs/README-develop-node-utils-APIs.md:56]: do we need a note about `git flow feature start` and/or just doing `git branch feature/FEATURE_NAME ; git checkout feature/FEATURE_NAME`?

File: docs/README-develop-node-utils-APIs.md [hunk 4]
```diff
@@ -126,14 +142,14 @@ end
 
 ### <a name="minitest">Step 3. Minitest: feature bash-shell</a>
 
-* A minitest should be created to validate the new APIs. Minitests are stored in the tests directory: `cisco_network_node_utils/tests/`
+* A minitest should be created to validate the new APIs. Minitests are stored in the tests directory: `cisco-network-node-utils/tests/`
```
Comment (glennmatthews) [inline docs/README-develop-node-utils-APIs.md]: seems like for consistency this should be changed to simply `tests/`?

File: docs/README-develop-node-utils-APIs.md [hunk 5]
```diff
@@ -236,6 +252,7 @@ Finished tests in 4.975186s, 0.8040 tests/s, 0.4020 assertions/s.
 
 *Note. The minitest harness counts the helper methods as tests which is why the final tally shows 4 tests instead of just 2 tests.*
 
+
```
Comment (glennmatthews) [inline docs/README-develop-node-utils-APIs.md]: unnecessary?

File: docs/README-develop-node-utils-APIs.md [hunk 6]
```diff
@@ -272,7 +303,7 @@ Example:
 
 As with the earlier example, `router eigrp` will need YAML definitions in the common file:
 
-`cisco_network_node_utils/lib/cisco_node_utils/command_reference_common.yaml`
+`cisco-network-node-utils/lib/cisco_node_utils/command_reference_common.yaml`
```
Comment (glennmatthews) [inline docs/README-develop-node-utils-APIs.md]: just remove the prefix? `lib/cisco_node_utils/command_reference_common.yaml`. Same comment on next diff below.

File: docs/README-develop-node-utils-APIs.md [hunk 7]
```diff
@@ -316,13 +347,13 @@ eigrp:
...
 ```
-cisco_network_node_utils/lib/cisco_node_utils.rb
+cisco-network-node-utils/lib/cisco_node_utils.rb
 ```
 
 * The `template-router.rb` file provides a basic router API that we will use as the basis for `router_eigrp.rb`:
 
 ```bash
-cp  docs/template-router.rb  cisco_network_node_utils/router_eigrp.rb
+cp  docs/template-router.rb  cisco-network-node-utils/router_eigrp.rb
```
Comment (glennmatthews) [inline docs/README-develop-node-utils-APIs.md]: lib/cisco_node_utils/router_eigrp.rb

File: docs/README-develop-node-utils-APIs.md [hunk 8]
```diff
@@ -470,7 +501,7 @@ end
 * Use `template-test_router.rb` to build the minitest for `router_eigrp.rb`:
 
 ```
-cp  docs/template-test_router.rb  cisco_network_node_utils/tests/test_router_eigrp.rb
+cp  docs/template-test_router.rb  cisco-network-node-utils/tests/test_router_eigrp.rb
```
Comment (glennmatthews) [inline docs/README-develop-node-utils-APIs.md]: tests/test_router_eigrp.rb

File: docs/README-develop-node-utils-APIs.md (hunk 1 above)
Comment (mikewiebe) [inline docs/README-develop-node-utils-APIs.md]: Fair enough.  I changed it back to cisco_node_utils

File: docs/README-develop-node-utils-APIs.md (hunk 2 above)
Comment (mikewiebe) [inline docs/README-develop-node-utils-APIs.md:38]: I thought about adding git-flow information but decided against it.  IMO, this is beyond the scope of this document.  Once they fork the repo, they can certainly use git-flow or any other preferred branch management tooling.  I could add a reference to git-flow suggesting they use it to manage their fork?

Thanks for the reminder about working in an environment where they don't have admin/root privileges.  In this case they will, but I will add the `gem install --user` but along with the `PATH` info.

File: docs/README-develop-node-utils-APIs.md (hunk 3 above)
Comment (mikewiebe) [inline docs/README-develop-node-utils-APIs.md:56]: Let's chat about how this would work.

File: docs/README-develop-node-utils-APIs.md (hunk 4 above)
Comment (mikewiebe) [inline docs/README-develop-node-utils-APIs.md]: Yeah, I agree.

File: docs/README-develop-node-utils-APIs.md (hunk 5 above)
Comment (mikewiebe) [inline docs/README-develop-node-utils-APIs.md]: Yes, removed.

File: docs/README-develop-node-utils-APIs.md (hunk 6 above)
Comment (mikewiebe) [inline docs/README-develop-node-utils-APIs.md]: Updated.

File: docs/README-develop-node-utils-APIs.md (hunk 7 above)
Comment (mikewiebe) [inline docs/README-develop-node-utils-APIs.md]: Updated

File: docs/README-develop-node-utils-APIs.md (hunk 8 above)
Comment (mikewiebe) [inline docs/README-develop-node-utils-APIs.md]: Updated

File: docs/README-develop-node-utils-APIs.md [hunk 9]
```diff
@@ -3,36 +3,81 @@
...
+This development guide uses tools that are packaged as gems that need to be installed on your server.
 
 ```bash
-git clone https://github.com/cisco/cisco-network-node-utils.git
+gem install cisco_nxapi
+gem install rubocop
+gem install minitest --version 4.3.2
+```
+
+**NOTE:** If you are working from a server where you don't have admin/root privilages, use the following commands to install the gems and then update your `PATH` to include `~/.gem/ruby/x.x.x/bin`
```
Comment (chrisvanheuveln) [inline docs/README-develop-node-utils-APIs.md]: /privileges/

File: docs/README-develop-node-utils-APIs.md [hunk 10]
```diff
@@ -3,36 +3,81 @@
...
+
+```bash
+git clone https://github.com/YOUR-USERNAME/cisco-network-node-utils.git
+cd cisco-network-node-utils/
+```
+
+As a best practice go ahead and create a topic/feature branch for your feature work using the `git branch feature/<feature_name>` command.
+
+```bash
+git branch feature/bash-shell
```
Comment (chrisvanheuveln) [inline docs/README-develop-node-utils-APIs.md]: oof. I changed all of the bash-shell references to 'tunnel' so you're going to have a bunch of merge conflicts.

**Pull request merged**